domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/BankAccount.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.cmc.domain.models.statementofmeans;
+
+import lombok.Builder;
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import uk.gov.hmcts.cmc.domain.constraints.Money;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+import javax.validation.constraints.NotNull;
+
+import static uk.gov.hmcts.cmc.domain.utils.ToStringStyle.ourStyle;
+
+@Builder
+public class BankAccount {
+
+    public enum BankAccountType {
+        CURRENT_ACCOUNT("Current account"),
+        SAVINGS_ACCOUNT("Savings account"),
+        ISA("ISA"),
+        OTHER("Other");
+
+        String description;
+
+        BankAccountType(String description) {
+            this.description = description;
+        }
+
+        public String getDescription() {
+            return this.description;
+        }
+    }
+
+    @NotNull
+    private final BankAccountType type;
+
+    private final boolean joint;
```

Comment:
Shouldn't we still make it `@NotNull` despite the fact that type does not allow nulls? I wonder what would have happened is somebody missed that filed. We would like to react with 400 and validation error and not some 500 due to the fact that application tries to set null to boolean field.

Reply:
`@NotNull` is meaningless on primitives, either we use `Boolean` and put `@NotNull`